src/Dialogs/UninstallFailDialog.vala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 elementary, Inc. (https://elementary.io)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+public class UninstallFailDialog : Granite.MessageDialog {
+    public AppCenterCore.Package package { get; construct; }
+    public Error error { get; construct; }
+
+    public UninstallFailDialog (AppCenterCore.Package package, Error error) {
+        Object (
+            title: "",
+            primary_text: _("Uninstall failed for %s").printf (package.get_name ()),
+            secondary_text: _("This may have been caused by external, manually added software repositories or a corrupted sources file."),
```

Comment:
I'm not sure that an uninstall failing would be caused by "manually added software repositories or corrupted sources files". Those type of errors occur when we're trying to update the local package cache from the repos, but shouldn't happen when trying to uninstall something.

Reply:
@davidmhewitt I think that copy just came verbatim from the other error dialog. :smile: Do we know what might _actually_ cause a failure when uninstalling?

Reply:
@cassidyjames I copied it from the other dialog. Couldn't think of anything better. :smile:  The actual error is shown in the error expander, whenever available. The sub heading was supposed to be a generic description. I am open to suggestions here.

Reply:
I have changes the string. Is it acceptable now?